### PR TITLE
[MRG] Add copy and channel selection for a Layout object

### PR DESCRIPTION
**Pull request by mscheltienne**
I had a report that `plot_evoked_topo(..., layout=layout, exclude=["M1", "M2"])` was partially ignoring the exclusion. The plot still had the 2 empty M1/M2 channels because they were present in the layout.

I added 2 methods to the `Layout` object: `copy` and `pick`. Nothing fancy, beside that the `pick` method is *not* based on a function in `mne._fiff.pick`. I did not find one that was fitting all the needs for a `Layout` instance. The `pick` method is now used in the plotting function.

### CODE REVIEW

**Comment by mscheltienne** [comment]
Renders: 
- `copy`: https://output.circle-artifacts.com/output/job/3ecee0f1-a728-4406-abde-a9b16d59a7dc/artifacts/0/html/generated/mne.channels.Layout.html#mne.channels.Layout.copy
- `pick`: https://output.circle-artifacts.com/output/job/3ecee0f1-a728-4406-abde-a9b16d59a7dc/artifacts/0/html/generated/mne.channels.Layout.html#mne.channels.Layout.pick

File: mne/channels/layout.py [hunk 1]
```diff
@@ -135,6 +154,105 @@ def plot(self, picks=None, show_axes=False, show=True):
...
+                    "The channel selection yielded no remaining channels. Please edit "
+                    "the arguments 'picks' and 'exclude' to include at least one "
+                    "channel."
+                )
+        else:
+            picks = np.array(list(set(picks)))
+        self.pos = self.pos[picks]
+        self.ids = self.ids[picks]
+        self.names = [self.names[k] for k in picks]
+        return self
```
Comment (agramfort) [inline mne/channels/layout.py:254]: did you check if this logic is not present elsewhere? I would be surprised if it's not.

Comment (mscheltienne) [reply mne/channels/layout.py:256]: Yes, the only oneI found that would work is `picks = _picks_to_idx(len(layout.names), picks)` which picks on a number of channels. It's more limited and restrictive on the inputs than the logic above.
I do plan to propose a new channel selection API, to try to 1. clean-up all the redundant code and multiple pick functions in [mne._fiff.pick](https://github.com/mne-tools/mne-python/blob/main/mne/_fiff/pick.py) and 2. open a public API for channel selection (#11913). Hopefully next week 😉

**Comment by agramfort** [comment]
>
> Yes, the only oneI found that would work is picks =
> _picks_to_idx(len(layout.names), picks) which picks on a number of
> channels. It's more limited and restrictive on the inputs than the logic
> above.
> I do plan to propose a new channel selection API, to try to 1. clean-up
> all the redundant code and multiple pick functions in mne._fiff.pick
> <https://github.com/mne-tools/mne-python/blob/main/mne/_fiff/pick.py> and
> 2. open a public API for channel selection (#11913
> <https://github.com/mne-tools/mne-python/issues/11913>). Hopefully next
> week 😉
>

yes I think #11913 should be done first here. I would not have this complex
and common code here.

**Comment by mscheltienne** [comment]
Agree, but I think #11913 will take time, thus I would first merge this PR with a TODO comment left in the codebase and a x-ref to replace the logic with whatever is defined in #11913. Especially as I do have someone at our site that could use this channel selection on a layout to improve his topographic plots.

**Review by larsoner — APPROVED** [review]
@mscheltienne still want this one? Sorry it fell off the radar a bit. Seems good enough to go for me

**Comment by mscheltienne** [comment]
It would be nice, yes :)

**Comment by larsoner** [comment]
Thanks @mscheltienne !

**Pull request merged**